script.xbmc.unpausejumpback/resources/lib/common.py

```diff
@@ -1,117 +1,199 @@
+    xbmc = None
+    unit_testing = True
+    KODI_VERSION = 'N/A'
 
+    print("\nNo user agent, must be unit testing.\n")
 
-def log_info(message, exception_instance=None):
-    """
-    Log a message at the LOGINFO level, i.e. even if Kodi debugging is not turned on. Use sparingly.
+    def log(message, exception_instance=None, level=None):
+        print(f'DEBUG: {message}')
+        if exception_instance:
+            print(f'EXCPT: {traceback.format_exc(exception_instance)}')
 
-    :param message: required, the message to log
-    :param exception_instance: optional, an instance of some Exception
-    """
+    def log_info(message, exception_instance=None):
+        log(f'INFO : {message}')
+        if exception_instance:
+            print(f'EXCPT: {traceback.format_exc(exception_instance)}')
 
-    log(message, exception_instance, level=xbmc.LOGINFO)
+else:
 
+    def log(message, exception_instance=None, level=xbmc.LOGDEBUG):
+        """
+        Log a message to the Kodi debug log, if debug logging is turned on.
 
-def footprints(startup=True):
-    """
-    Log the startup of an addon, and key Kodi details that are helpful for debugging
+        :param message: required, the message to log
+        :param exception_instance: optional, an instance of some Exception
+        :param level: optional, the Kodi log level to use, default LOGDEBUG
+        """
 
-    :param startup: optional, default True.  If true, log the startup of an addon, otherwise log the exit.
-    """
-    if startup:
-        log_info(f'Starting...')
-        log_info(f'Kodi Version: {KODI_VERSION}')
-        log_info(f'Addon arguments: {ADDON_ARGUMENTS}')
-    else:
-        log_info(f'Exiting...')
+        message = f'### {ADDON_NAME} {ADDON_VERSION} - {message}'
+        message_with_exception = message + f' ### Exception: {traceback.format_exc(exception_instance)}'
+
+        if exception_instance is None:
+            xbmc.log(message, level)
+        else:
+            xbmc.log(message_with_exception, level)
 
 
-def set_property(window, name, value=""):
-    """
-    Set a property on a window.
-    To clear a property, provide an empty string
+    def log_info(message, exception_instance=None):
+        """
+        Log a message at the LOGINFO level, i.e. even if Kodi debugging is not turned on. Use very sparingly.
 
-    :param window: Required.  The Kodi window on which to set the property.
-    :param name: Required.  Name of the property.
-    :param value: Optional (defaults to "").  Set the property to this value.  An empty string clears the property.
-    """
-    window.setProperty(name, value)
-    # if value and value != False and value != 'na.png':
-    #     log(f'Set window property: [{name}] - value: [{value}]')
+        :param message: required, the message to log
+        :param exception_instance: optional, an instance of some Exception
+        """
+        log(message, exception_instance, level=xbmc.LOGINFO)
```

Comment:
default log level should be debug

Reply:
> default log level should be debug

...it is.  This is a line in a function specific to log at the info level (rarely used) - the default log function just above is the one used almost always by my addons, and it default to only log at the debug levl.  This is just for circumstances where its' _really_ useful to see something being logged. 

Is that ok?   We've actually had this chat before and it was ok then, but I guess I could move _everything_ to debug log, but I'd prefer not to.

Reply:
Yeah it is fine. The root cause of myself bringing this all the time is the way the function is written it just implies info is the default level when you already have the other function where you can easily override the log level when needed. I advise you to drop this method in the future and call log with level=info in the caller's if possible to avoid bringing the same comment again :)